src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/DefaultCache.java

```diff
@@ -0,0 +1,244 @@
+            String newEtag = cacheEntry.etag();
+            assert status != null;
+
+            if (ifNoneMatch != null && newEtag != null)
+            {
+                return !(status.equals(HttpStatus.OK_200) && isMatchByEtag(requestHeaders, newEtag));
+            }
+        }
+
+        return true;
+    }
+
+    public boolean isUpdatedByResponseHeadersToRetry(
+        ListFW<HttpHeaderFW> requestHeaders,
+        ListFW<HttpHeaderFW> responseHeaders,
+        String ifNoneMatch,
+        int requestHash)
+    {
+
+        if (isPreferWait(requestHeaders)
+            && !isPreferenceApplied(responseHeaders))
+        {
+            String status = HttpHeadersUtil.getHeader(responseHeaders, HttpHeaders.STATUS);
+            String newEtag = getHeader(responseHeaders, ETAG);
+            boolean etagMatches = false;
+            assert status != null;
+
+            if (ifNoneMatch != null && newEtag != null)
+            {
+                etagMatches = status.equals(HttpStatus.OK_200) && isMatchByEtag(requestHeaders, newEtag);
+
+                if (etagMatches)
+                {
+                    DefaultCacheEntry cacheEntry = cachedEntries.get(requestHash);
+                    cacheEntry.updateResponseHeader(responseHeaders);
+                }
+            }
+
+            boolean notModified = status.equals(NOT_MODIFIED_304) || etagMatches;
+
+            return !notModified;
+        }
+
+        return true;
+    }
+
+    public void send304(
+        DefaultCacheEntry entry,
+        ListFW<HttpHeaderFW> requestHeaders,
+        MessageConsumer acceptReply,
+        long acceptRouteId,
+        long acceptReplyId)
+    {
+        if (DEBUG)
+        {
+            System.out.printf("[%016x] ACCEPT %016x %s [sent response]\n",
+                              currentTimeMillis(), acceptReplyId, "304");
+        }
+
+        if (isPreferIfNoneMatch(requestHeaders))
+        {
+            String preferWait = getHeader(requestHeaders, PREFER);
+            String ifNonMatch = getHeader(requestHeaders, IF_NONE_MATCH);
+            writer.doHttpResponse(acceptReply,
+                                          acceptRouteId,
+                                          acceptReplyId,
+                                          supplyTrace.getAsLong(),
+                                          e -> e.item(h -> h.name(STATUS).value(NOT_MODIFIED_304))
+                                                .item(h -> h.name(ETAG).value(ifNonMatch))
+                                                .item(h -> h.name(PREFERENCE_APPLIED).value(preferWait)));
+        }
+        else
+        {
+            writer.doHttpResponse(acceptReply,
+                                          acceptRouteId,
+                                          acceptReplyId,
+                                          supplyTrace.getAsLong(),
+                                          e -> e.item(h -> h.name(STATUS).value("304"))
```

Comment:
use constant `NOT_MODIFIED_304` instead of `"304"`

Reply:
Done.